ext/spl/spl_iterators.stub.php

```diff
@@ -0,0 +1,475 @@
+    public function current() {}
+
+    /** @return void */
+    public function next() {}
+}
+
+abstract class FilterIterator extends IteratorIterator
+{
+    /** @return bool */
+    abstract public function accept();
+
+    public function __construct(Iterator $iterator) {}
+
+    /** @return void */
+    public function rewind() {}
+
+    /** @return bool */
+    public function valid() {}
+
+    /** @return mixed */
+    public function key() {}
+
+    /** @return mixed */
+    public function current() {}
+
+    /** @return void */
+    public function next() {}
+
+    /** @return Iterator|null */
+    public function getInnerIterator() {}
+}
+
+abstract class RecursiveFilterIterator extends FilterIterator implements RecursiveIterator
+{
+    public function __construct(RecursiveIterator $iterator) {}
+
+    /** @return bool */
+    public function hasChildren() {}
+
+    /** @return RecursiveFilterIterator|null */
+    public function getChildren() {}
+}
+
+class ParentIterator extends RecursiveFilterIterator
+{
+    public function __construct(RecursiveIterator $iterator) {}
+
+    /** @return bool */
+    public function accept() {}
+}
+
+interface SeekableIterator extends Iterator
+{
+    /** @return void */
+    public function seek(int $position);
+}
+
+class LimitIterator extends IteratorIterator
+{
+    public function __construct(Iterator $iterator, int $offset = 0, int $count = -1) {}
+
+    /** @return void */
+    public function rewind() {}
+
+    /** @return bool */
+    public function valid() {}
+
+    /** @return mixed */
+    public function key() {}
+
+    /** @return mixed */
+    public function current() {}
+
+    /** @return void */
+    public function next() {}
+
+    /** @return int */
+    public function seek(int $position) {}
+
+    /** @return int */
+    public function getPosition() {}
+
+    /** @return Iterator|null */
+    public function getInnerIterator() {}
+}
+
+class CachingIterator extends IteratorIterator implements ArrayAccess, Countable
+{
+    public function __construct(Iterator $iterator, int $flags = self::CALL_TOSTRING) {}
+
+    /** @return void */
+    public function rewind() {}
+
+    /** @return bool */
+    public function valid() {}
+
+    /** @return mixed */
+    public function key() {}
+
+    /** @return mixed */
+    public function current() {}
+
+    /** @return void */
+    public function next() {}
+
+    /** @return bool */
+    public function hasNext() {}
+
+    public function __toString(): string {}
+
+    /** @return Iterator|null */
+    public function getInnerIterator() {}
+
+    /** @return int */
+    public function getFlags() {}
+
+    /** @return void */
+    public function setFlags(int $flags) {}
+
+    /**
+     * @param string $index
+     * @return mixed
+     */
+    public function offsetGet($index) {}
+
+    /**
+     * @param string $index
+     * @param mixed $value
+     * @return void
+     */
+    public function offsetSet($index, $value) {}
```

Comment:
```suggestion
    public function offsetSet(string $index, $value) {}
```

Reply:
Actually, I had to remove the `string` type from here because the signature wasn't compatible with `ArrayAccess`.

Reply:
Ugh... you're right :(

Reply:
Yep, I was also a bit sad